src/btree/bt_sync.c

```diff
@@ -221,8 +313,40 @@ __sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
 				leaf_bytes += page->memory_footprint;
 				++leaf_pages;
 			}
+
+			/*
+			 * If the page needs forced eviction, try to do that
+			 * now.
+			 *
+			 * For eviction to have a chance, we first need to move
+			 * the walk point to the next page checkpoint will
+			 * visit.  We want to avoid this code being too special
+			 * purpose, so try to reuse the ordinary eviction path.
+			 *
+			 * If eviction succeeded, it steps to the next ref, so
+			 * we have to skip the next walk.  If eviction fails,
+			 * remember so we don't retry it.
+			 */
+			if (!WT_PAGE_IS_INTERNAL(page) &&
+			    page->read_gen == WT_READGEN_OLDEST &&
+			    !evict_failed) {
```

Comment:
One additional test here might be to test `page->modify->rec_result != WT_PM_REC_MULTIBLOCK`, because we can't evict any page that might require a split, there's no point in trying.

Reply:
I'm inclined to leave that to eviction to figure out -- we should be able to allow splits in this path, so I'd prefer not to have extra checks preventing them.